packages/paste-core/components/inline-control-group/src/InlineControlGroup.tsx

```diff
@@ -73,13 +79,9 @@ const InlineControlGroup = React.forwardRef<HTMLFieldSetElement, InlineControlGr
                         : fieldStyleProps?.marginTop
                       : 'space40'
                   }
-                  marginRight={
-                    fieldStyleProps?.marginRight
-                      ? fieldStyleProps?.marginRight
-                      : // eslint-disable-next-line unicorn/no-nested-ternary
-                      orientation === 'horizontal'
-                      ? 'space70'
-                      : null
+                  marginRight={orientation === 'horizontal' && !isVisualPicker ? 'space70' : null}
+                  marginLeft={
+                    isVisualPicker && orientation === 'horizontal' ? (index === 0 ? 'space0' : 'space30') : 'space0' // Sets spacing between horizontal Visual Pickers except for the first one
```

Comment:
praise: you're right, I do dislike this, but the comments here, as to why its here, helps us reason about this and ultimately we might find time to refactor this out to a better primitive in the future.